infra/cifuzz/run_fuzzers.py

```diff
@@ -202,6 +209,55 @@ class BatchFuzzTargetRunner(BaseFuzzTargetRunner):
   def quit_on_bug_found(self):
     return False
 
+  def run_fuzz_target(self, fuzz_target_obj):
+    """Fuzzes with |fuzz_target_obj| and returns the result."""
+    result = fuzz_target_obj.fuzz()
+    logging.debug('corpus_path: %s', os.listdir(result.corpus_path))
+    self.clusterfuzz_deployment.upload_corpus(fuzz_target_obj.target_name,
+                                              result.corpus_path)
+    # This must be done after we upload the corpus, otherwise it will be deleted
+    # before we get a chance to upload it. We can't delete the fuzz target
+    # because it is needed when we upload the build.
+    fuzz_target_obj.free_disk_if_needed(delete_fuzz_target=False)
```

Comment:
There seems to be a lot of ordering assumptions tying all these different parts of the code together, making things a bit difficult to understand. Is there any way we can remove/clean some of these up? e.g. knowing that we can't delete the fuzz target here because somewhere else later it needs to be uploaded. these ideally should be more modular and independent of one another. 

Do we need to call `free_disk_if_needed` in `run_fuzz_target`?. It's a side effect that can be surprising to the caller. We can do this in the caller instead? Are there steps in between here and when we return to `run_fuzz_targets` that requires this extra disk space? 

i.e.

```python
... inside BaseFuzzTargetRunner
def run_fuzz_targets(self):
  ...
  for target_path in self.fuzz_target_paths:
    ...
     target = self.create_fuzz_target_obj(target_path, run_seconds)
     start_time = time.time()
     result = self.run_fuzz_target(target)
     target.cleanup()  # clean up here instead.
```

This might not work as is when you have this `delete_fuzz_target` logic.. but why do we need to do this in some cases by not others?

Reply:
I like this idea of .clean() and it will work with delete_fuzz_target if I override it.
We can't delete fuzz targets in batch fuzzing runs because batch fuzzing runs need to upload the target. CI fuzzing runs don't need to upload the target so for them it makes sense to delete.